signer/src/bitcoin/validation.rs

```diff
@@ -0,0 +1,434 @@
+    AssessedFeeTooHigh(OutPoint),
+    /// The signer is not part of the signer set that generated the
+    /// aggregate public key used to lock the deposit funds.
+    ///
+    /// TODO: For v1 every signer should be able to sign for all deposits,
+    /// but for v2 this will not be the case. So we'll need to decide
+    /// whether a particular deposit cannot be signed by a particular
+    /// signers means that the entire transaction is rejected from that
+    /// signer.
+    #[error("the signer is not part of the signing set for the aggregate public key; {0}")]
+    CannotSignUtxo(OutPoint),
+    /// The deposit transaction has been confirmed on a bitcoin block
+    /// that is not part of the canonical bitcoin blockchain.
+    #[error("deposit transaction not on canonical bitcoin blockchain; {0}")]
+    TxNotOnBestChain(OutPoint),
+    /// The deposit UTXO has already been spent.
+    #[error("deposit transaction used as input in confirmed sweep transaction; {0}")]
+    DepositUtxoSpent(OutPoint),
+    /// Given the current time and block height, it would be imprudent to
+    /// attempt to sweep in a deposit request with the given lock-time.
+    #[error("lock-time expiration is too soon; {0}")]
+    LockTimeExpiry(OutPoint),
+    /// The signer does not have a record of their vote on the deposit
+    /// request in their database.
+    #[error("the signer does not have a record of their vote on the deposit request; {0}")]
+    NoVote(OutPoint),
+    /// The signer has rejected the deposit request.
+    #[error("the signer has not accepted the deposit request; {0}")]
+    RejectedRequest(OutPoint),
+    /// The signer does not have a record of the deposit request in their
+    /// database.
+    #[error("the signer does not have a record of the deposit request; {0}")]
+    Unknown(OutPoint),
+    /// The locktime in the reclaim script is in time units and that is not
+    /// supported. This shouldn't happen, since we will not put it in our
+    /// database is this is the case.
+    #[error("the deposit locktime is denoted in time and that is not supported; {0}")]
+    UnsupportedLockTime(OutPoint),
+}
+
+/// The responses for validation of a sweep transaction on bitcoin.
+#[derive(Debug, thiserror::Error, PartialEq, Eq, Copy, Clone)]
+pub enum BitcoinSweepErrorMsg {
+    /// The error has something to do with the inputs.
+    #[error("the assessed fee for a deposit would exceed their max-fee")]
+    Deposit(#[from] BitcoinDepositInputError),
+}
+
+/// A struct for a bitcoin validation error containing all the necessary
+/// context.
+#[derive(Debug)]
+pub struct BitcoinValidationError {
+    /// The specific error that happened during validation.
+    pub error: BitcoinSweepErrorMsg,
+    /// The additional information that was used when trying to
+    /// validate the complete-deposit contract call. This includes the
+    /// public key of the signer that was attempting to generate the
+    /// `complete-deposit` transaction.
```

Comment:
nit: wrong copypasted comment

Reply:
🙏🏽

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/744/commits/1da998d1d1db6b789b67ee90d1ff08d57a170a0e